addons/pos_loyalty/static/src/js/Loyalty.js

```diff
@@ -185,18 +190,62 @@ const PosLoyaltyOrderline = (Orderline) => class PosLoyaltyOrderline extends Ord
                 }
             });
             for (const line of linesToRemove) {
-                this.order.orderlines.remove(line);
+                this.order.remove_orderline(line);
             }
         }
         return super.set_quantity(...arguments);
     }
+    /**
+     * @override
+     */
+    can_be_merged_with(otherLine) {
+        if (this.is_reward_line && otherLine.is_reward_line) {
+            if (otherLine.reward_id == this.reward_id) {
+                const reward = this.order.pos.reward_by_id[this.reward_id];
+                if (reward && reward.reward_type == 'product') {
```

Comment:
what if it's a multi_product reward but with the same product selected at the end?

Reply:
I take back what I said, I made a mistake thinking the `multi_product` value was stored in the `reward_type`